linkshell_service/linkshell_service.lua

```diff
@@ -0,0 +1,76 @@
+local packets = require('packets')
+local shared = require('shared')
+
+local linkshell = shared.new('linkshell')
+
+local linkshell.env = {
+    next = next,
+}
+  
+local linkshell.data = {
+    [1] = { lsmes = { }, color = { }, },
+    [2] = { lsmes = { }, },
+}
+
+local handle_color = function(p)
+    local data = linkshell.data[1]
+    data.color.red = p.linkshell1_red
+    data.color.green = p.linkshell1_green
+    data.color.blue = p.linkshell1_blue
+end
+
+packets.incoming[0x0CC]:register(function(p)
+    local ls_number = bit.band(p.flags, 0x40) == 0x40 and 2 or 1
+    local data = linkshell.data[ls_number]
+    data.name = p.linkshell_name
+    data.lsmes.timestamp = p.timestamp
+    data.lsmes.author = p.player_name
+    data.lsmes.text = p.message
+    data.lsmes.permissions = p.permissions
+end)
+
+packets.incoming[0x0E0]:register(function(p)
+    local data = linkshell.data[p.linkshell_number]
+    data.bag_index = p.bag_index
+end)
+
+packets.incoming[0x00D]:register(handle_color)
+packets.incoming[0x037]:register(handle_color)
+
+last_00D = packets.incoming[0x00D].last
+last_037 = packets.incoming[0x037].last
+
+if last_00D then
+    handle_color(last_00D)
+end
+if last_037 then
+    handle_color(last_00B)
+end
```

Comment:
I cannot imagine the `0x00D` packet being useful here. All LS info changes should be in `0x0CC` and `0x0E0` imo, while `0x037` also has LS info in it, I don't think it can ever be different or more useful than `0x0CC`. I would remove everything but those two, and adjust `last_XXX` accordingly. Also, make the last packets `local`.

This has the advantage that we also don't need to worry about the order, since they carry entirely separate information. I would say go this route until we notice errors with this approach, which I don't think there should be.